pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/AbstractJUnitRule.java

```diff
@@ -89,7 +89,7 @@ private boolean isJUnit3Method(ASTMethodDeclaration method) {
         return isJUnit3Class && method.isVoid() && method.getName().startsWith("test");
     }
 
-    private boolean isJUnit3Class(ASTCompilationUnit node) {
+    protected boolean isJUnit3Class(ASTCompilationUnit node) {
```

Comment:
I'd not expose these methods.
These methods are called already when visiting CompilationUnit and the result is stored in fields, see lines 38-40 above.
I'd make these fields protected, so that you can use it in the rule.

Reply:
Indeed, fixed.